### ISSUE TITLE
"There is no source with this ID" thrown when Source is removed quickly

### ISSUE THREAD

**Issue by fenech** [visgl/react-map-gl#1393]
#### Description

Rendering a `<Source>` and removing it quickly causes an exception to be thrown. This is because `map.removeSource(id)` is called before the source has been created.

A possible easy fix would be to just add a check `map.getSource(id)` to [the existing check](https://github.com/visgl/react-map-gl/blob/bdddc49805f2bc51916864bda21bdbf7758ba208/src/components/source.js#L117).

Happy to open a PR with that if you think it's a good idea!

#### Repro Steps

```js
import { useEffect, useState } from "react";
import InteractiveMap, { Source } from "react-map-gl";

export default function App() {
  const [showSource, setShowSource] = useState(true);
  useEffect(() => {
    requestAnimationFrame(() => setShowSource(false));
  }, []);

  return (
    <InteractiveMap
      height={100}
      width={100}
      mapboxApiAccessToken="API TOKEN"
    >
      {showSource && <Source type="geojson" data={{}}/>}
    </InteractiveMap>
  );
}
```

#### Environment (please complete the following information):

- Library Version: react-map-gl 6.1.10
- Mapbox Version: mapbox-gl 2.1.1
- React Version: react 17.0.1
- Browser Version: Firefox 86.0
- OS: GNU/Linux Fedora 33

### There is no source with this ID

removeSource
https://tg2bo.csb.app/node_modules/mapbox-gl/dist/mapbox-gl.js:17999:34
removeSource
https://tg2bo.csb.app/node_modules/mapbox-gl/dist/mapbox-gl.js:23614:30
Source/</</<
https://tg2bo.csb.app/node_modules/react-map-gl/dist/esm/components/source.js:125:17

**Comment by VictorVelarde** [comment]
Hi @fenech,

Thx for your report. I think adding the check is a nice fix, would you like to open a PR?

**Comment by VictorVelarde** [comment]
This should be fixed with latest releases: 6.1.11 / 5.3.11. @fenech please confirm everything is fine on your side, and we'll close this issue

**Comment by fenech** [comment]
I can confirm that it's fixed in 6.1.11, thanks!
